packages/plugin-restify/src/restify.js

```diff
@@ -0,0 +1,74 @@
+const domain = require('domain') // eslint-disable-line
+const extractRequestInfo = require('./request-info')
+const createReportFromErr = require('@bugsnag/core/lib/report-from-error')
+const handledState = {
+  severity: 'error',
+  unhandled: true,
+  severityReason: {
+    type: 'unhandledErrorMiddleware',
+    attributes: { framework: 'Restify' }
+  }
+}
+
+module.exports = {
+  name: 'restify',
+  init: client => {
+    const requestHandler = (req, res, next) => {
+      const dom = domain.create()
+
+      // Start a session whether sessions are used or not. We use this
+      // to store request-specific info, in case of any errors.
+      const sessionClient = client.startSession()
+
+      // attach it to the request
+      req.bugsnag = sessionClient
+
+      // extract request info and pass it to the relevant bugsnag properties
+      const requestInfo = extractRequestInfo(req)
+      sessionClient.metaData = { ...sessionClient.metaData, request: requestInfo }
+      sessionClient.request = {
+        clientIp: requestInfo.clientIp,
+        headers: requestInfo.headers,
+        httpMethod: requestInfo.httpMethod,
+        url: requestInfo.url,
+        referer: requestInfo.referer
+      }
+
+      // unhandled errors caused by this request
+      dom.on('error', (err) => {
+        req.bugsnag.notify(createReportFromErr(err, handledState), {}, (e, report) => {
+          if (e) return client._logger('Failed to send report to Bugsnag')
+          req.bugsnag.config.onUncaughtException(err, report, client._logger)
+        })
+        if (!res.headersSent) {
+          const body = 'Internal server error'
+          res.writeHead(200, {
+            'Content-Length': Buffer.byteLength(body),
+            'Content-Type': 'text/plain'
+          })
+          res.end(body)
+        }
+      })
+
+      return dom.run(next)
+    }
+
+    const errorHandler = (req, res, err, cb) => {
+      if (err.statusCode && err.statusCode < 500) return cb()
```

Comment:
what situation will the error handler be called where we don't want to pass the `err` to the callback?

Reply:
> what situation will the error handler be called

I don't actually know. It's not apparent from restify's docs and I couldn't glean it from the amount I looked at the code. I _think_ passing an error to the `cb()` would indicate that an error happend _within_ this function.